src/Workspaces/Core/Portable/FlowAnalysis/LValueFlowCaptureProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+    /// <summary>
+    /// Helper class to detect <see cref="IFlowCaptureOperation"/>s that are l-value captures.
+    /// </summary>
+    internal static class LValueFlowCapturesProvider
```

Comment:
>LValueFlowCapturesProvider [](start = 26, length = 26)

@AlekseyTs @333fred This is the helper class that I needed to use to identify l-value flow captures so we can correctly detect read/writes in a control flow graph. This computation needs an additional pass over the entire flow graph, and is susceptible to break in future if we tweak/expand how we generate l-value flow captures in flow graph..It would be good if we can instead expose this information directly on the IFlowCaptureOperation/IFlowCaptureReferenceOperation. I will file a tracking bug for this work.

Reply:
Filed #31007

---
In reply to: [228555514](https://github.com/dotnet/roslyn/pull/30777#discussion_r228555514) [](ancestors = 228555514)